SpatialGDK/Source/SpatialGDK/Private/EngineClasses/Components/ActorInterestComponent.cpp

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "EngineClasses/Components/ActorInterestComponent.h"
+
+#include "Schema/Interest.h"
+#include "Interop/SpatialClassInfoManager.h"
+
+void UActorInterestComponent::CreateQueries(const USpatialClassInfoManager& ClassInfoManager, const SpatialGDK::QueryConstraint& AdditionalConstraints, TArray<SpatialGDK::Query>& OutQueries) const
+{
+	for (const auto& QueryData : Queries)
+	{
+		if (!QueryData.Constraint)
+		{
+			continue;
+		}
+
+		SpatialGDK::Query NewQuery{};
+		if (AdditionalConstraints.IsValid())
```

Comment:
Why are component constraints only added if AdditionalConstraints are specified? Could you add a comment.

Reply:
it's added either way, the guard is to prevent the redundant AND constraint. Added a comment.